File: pw/branches/r1117/Tools/ReimportTerrain.py
#!python -w

import os
import os.path
import re
import string

dirname, root = map(os.path.abspath, [r'..\Data\Terrain\Elements', r'..\Data'])
root = re.compile(re.sub(r'\\', r'\\\\', root + '\\'))

result = []
for name in os.listdir(dirname):
  if re.compile(".stat.xdb", re.IGNORECASE).search(name):
    path = os.path.join(dirname, name)
    result.append(root.sub('', path))

os.system(r'cd ..\Bin && ..\Src\_Debug\PF_EditorC.exe import --fast ' + string.join(result, ' '))

File: pw/branches/r1117/Tools/ReimportTerrain.rb
#!ruby -w

dirname, root = ['..\Data\Terrain\Elements', '..\Data'].collect { |dir| File.expand_path(dir) }
root = Regexp.new(root + '/')

result = Array.new
Dir.foreach(dirname) do |name|
  if name =~ %r".stat.xdb"i
    path = File.join(dirname, name)
    result << path.sub(root, '')
  end
end

system("cd ..\\Bin && ..\\Src\\_Debug\\PF_EditorC.exe import --fast #{result.join(' ')}")
